Python/Product/PyDebugAttach/PyDebugAttach.cpp

```diff
@@ -406,7 +407,7 @@ char* ReadCodeFromFile(wchar_t* filePath) {
     auto length = filestr.tellg();
     filestr.seekg(0, ios::beg);
 
-    int len = (int)length;
+    SSIZE_T len = (int)length;
```

Comment:
![image](https://user-images.githubusercontent.com/3315992/53908872-359ce680-4005-11e9-89e6-a95b08feeb34.png)

Reply:
What's the operator * it's complaining about? I don't see any multiplications here.

As changed, this adds a potential integer overflow in the for-loop below, because the index variable is compared `<= len`, but is itself `int`, so it'll loop forever. The original code is broken for files >sizeof(int), but only in a sense that it would not load the complete file - it would at least return.

(This is all strictly hypothetical, since we're loading specific files here that we know are small enough that `int` is fine in practice. Maybe this just needs a check-and-error-return after the line that assigns `length`? Then the rest of it doesn't need to worry about dealing with size mismatches, overflows etc. The types here can be tricky to reconcile properly, because there's no guarantee that `streamoff` is not larger than `ssize_t`, so doing this strictly correctly requires a great deal more checks.)

Reply:
Although given what this code really does, I think it could be made much simpler if it doesn't try to micro-optimize for allocations (which it shouldn't, because this isn't a hot path). If the function is changed to return `std::string`, then the entire thing after opening the file can be replaced with:
```cpp
std::string s(std::istreambuf_iterator<char>(filestr), {});
s.erase(std::remove(s.begin(), s.end(), '\r'), str.end());
return s;
```
And the caller can call `data()` explicitly to get the raw pointer.

Reply:
Or alternatively:
```cpp
std::string s;
std::copy_if(
    std::istreambuf_iterator<char>(filestr), {},
    std::back_inserter(s),
    [](auto ch) { return ch != '\r'; }
);
```

Reply:
Updated in latest commit